### PR TITLE
Use Toolbar API instead of BrowserToolbar in CustomTabsToolbarFeature

### LINKED ISSUE #2103: feature-customtabs: Depend on concept-toolbar
Ideally the feature component shouldn't depend on an actual implementation of `concept-toolbar`. However this means `concept-toolbar` would need to offer more options for customization. I wouldn't want to copy all the public API for that from `browser-toolbar` to `concept-toolbar` though.

### LINKED ISSUE #6261: Create new concept component for menu
We have several use cases where our current `browser-menu` component isn't compatible with our architecture:
 - We want to add menus to other APIs.
 - Multiple menu implementations that inherit from each other but do not use a common API.

We should introduce a new component `concept-menu` that moves the `browser-menu` APIs to it's own place.

### PR DESCRIPTION
**Pull request by jonalmeida**
Creates a new concept-menu component and updates the Toolbar API to be in line with what we want to be able to do with it via Custom Tabs.

---

See individual commits for breakdown; will be squashed before merging.

Close #6261, close #2103.

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- [ ] **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Comment by jonalmeida** [comment]
That ended up being a lot more work just to be able to do this. 😄 

Screen Shot 2020-03-24 at 6 25 07 PM

File: components/concept/menu/src/main/java/mozilla/components/concept/menu/Menu.kt [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+     * Display the menu.
+     *
+     * @param anchor the view on which to pin the popup window.
+     * @param orientation the preferred orientation to show the popup window.
+     * @param endOfMenuAlwaysVisible when is set to true makes sure the bottom of the menu is always visible otherwise,
+     *  the top of the menu is always visible.
+     */
+    fun show(
+        anchor: View,
+        orientation: Orientation = Orientation.DOWN,
```
Comment (jonalmeida) [inline components/concept/menu/src/main/java/mozilla/components/concept/menu/Menu.kt]: These defaults should be on the `BrowserMenu` impl only, not on the interface.

**Comment by jonalmeida** [comment]
> As this is a significant change, I would like to do some manual testing.
There are some compiling issue, they are listed on the TC failures.

The sample browser should still compile and was were I did my testing. I'll fix the breaking APIs in the other samples shortly.

**Comment by jonalmeida** [comment]
@Amejia481 I've fixed the broken tests and sample-toolbar. Will update the changelog with the breaking APIs if this a direction we'd like to go.

File: samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt [hunk 2]
```diff
@@ -152,7 +152,7 @@ class ToolbarActivity : AppCompatActivity() {
         val open = SimpleBrowserMenuItem("Open in…") { /* Do nothing */ }
         val settings = SimpleBrowserMenuItem("Settings") { /* Do nothing */ }
 
-        val builder = BrowserMenuBuilder(listOf(fenix, share, homeScreen, open, settings))
+        val builder = BrowserMenuBuilder(mutableListOf(fenix, share, homeScreen, open, settings))
```
Comment (Amejia481) [inline samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:155]: should we mark this in the changelog?

**Comment by jonalmeida** [comment]
Looking at this again, I'm quite unhappy with what I've done here. The mutableList and mutableMap for the `MenuBuilder` is a step back from what we have.

**Comment by jonalmeida** [comment]
Closing for now, but this is still a change I'd like to land (without the menu updates).

**Pull request closed without merging**